src/uiScreens.c

```diff
@@ -350,11 +350,11 @@ void ui_displayStakingInfoScreen(
 			heading = STAKING_HEADING_WARNING;
 			strncpy(stakingInfo, "legacy Byron address (no staking rewards)", SIZEOF(stakingInfo));
 
-		} else if (addressParams->type == ENTERPRISE) {
+		} else if (addressParams->type == ENTERPRISE_KEY || addressParams->type == ENTERPRISE_SCRIPT) {
```

Comment:
This long if looks like a good candidate for refactoring into a switch statement.